content/300-guides/200-deployment/04-deploying-to-azure-functions.mdx

```diff
@@ -0,0 +1,444 @@
+
+Run the following command, after replacing `FUNCTION_APP_NAME_FROM_STEP_8` with the name of the **Function App** created in step 8:
+
+```terminal
+az functionapp config appsettings set --name FUNCTION_APP_NAME_FROM_STEP_8 \
+--resource-group prisma-azure-example \
+--settings DATABASE_URL=$DATABASE_URL
+```
+
+The command will set the `DATABASE_URL` app setting using the locally defined `DATABASE_URL` environment variable set in step 9.
+
+> **Note:** By default, app settings are not encrypted. Since the `DATABASE_URL` contains sensitive information, it can be stored more securely using [Azure's Key Vault](https://docs.microsoft.com/en-us/azure/app-service/app-service-key-vault-references?toc=/azure/azure-functions/toc.json)
+
+Congratulations! You have created all the necessary resources and configuration, which means your API is ready to be deployed.
+
+## 11. Deploy the functions
+
+In this step, you will generate Prisma Client and deploy the functions.
+
+
+From the project folder, run the following command:
+
+```terminal
+npx prisma generate
+```
+
+The command will generate Prisma Client into the `node_modules` folder.
+
+To deploy the functions, run the following command:
+
+```terminal
+func azure functionapp publish prisma-azure-example
+```
+
+> **Checkpoint:** If the functions have been successfully deployed you should see the following:
+
+```terminal
+$ func azure functionapp publish prisma-azure-exampl
+Getting site publishing info...
+Uploading package...
+Uploading 67.24 MB [##############################################################################]
+Upload completed successfully.
+Deployment completed successfully.
+Syncing triggers...
+Functions in APP_NAME:
+    CreatePost - [httpTrigger]
+        Invoke url: https://APP_NAME.azurewebsites.net/api/post
+    CreateUser - [httpTrigger]
+        Invoke url: https://APP_NAME.azurewebsites.net/api/user
+    DeletePost - [httpTrigger]
+        Invoke url: https://APP_NAME.azurewebsites.net/api/post/{postid}
+    FilterPosts - [httpTrigger]
+        Invoke url: https://APP_NAME.azurewebsites.net/api/filterposts
+    GetFeed - [httpTrigger]
+        Invoke url: https://APP_NAME.azurewebsites.net/api/feed
+    GetPost - [httpTrigger]
+        Invoke url: https://APP_NAME.azurewebsites.net/api/post/{postid}
+    PublishPost - [httpTrigger]
+        Invoke url: https://APP_NAME.azurewebsites.net/api/publish/{postid}
+```
+
+Congratulations 🎊! If you've made it this far, you've successfully deployed a Prisma based REST API to Azure Functions which uses Azure SQL as the database.
+
+In the next step, you'll test the functions and take a closer look at how the functions are implemented.
+
+## 12. Test the deployed functions
```

Comment:
Did you get this error in the response from the API?

It seems that the IP in the error is a Microsoft one (see screenshot below). This might mean that there's a problem connecting the functions to the database which is strange because this should typically not require any special firewall rules.

<img width="547" alt="Screen Shot 2021-03-04 at 14 57 03" src="https://user-images.githubusercontent.com/1992255/109974465-e4aa5080-7cf9-11eb-9fec-43ac8e4d929b.png">

Reply:
>Did you get this error in the response from the API?

Yes

Reply:
I flipped this setting on the Azure portal (firewall setting), and it works now
![image](https://user-images.githubusercontent.com/33921841/109975718-fd236680-7d0b-11eb-8f81-ed725b25e15d.png)

Reply:
Well spotted! 

I found this in the docs: https://docs.microsoft.com/en-us/azure/azure-sql/database/firewall-configure#connections-from-inside-azure

This can be done with the Azure CLI using 
```
az sql server firewall-rule create --resource-group prisma-azure-example --server UNIQUE_DB_SERVER_NAME --name allow-function-acccess --start-ip-address 0.0.0.0 --end-ip-address 0.0.0.0
```

I just tested and this effectively flips that setting in the UI.

Reply:
Could you double check that the command works for you too?

Reply:
That fixes it! 🎉